apps/cloud_inference/server_app/source/inference_hip_rocal.cpp

```diff
@@ -0,0 +1,1092 @@
+                                int bb_info = (numBB_per_message & 0xFFFF) | (numBB << 16);
+                                InfComCommand cmd = {
+                                    INFCOM_MAGIC, INFCOM_CMD_BB_INFERENCE_RESULT, { tag, bb_info }, { 0 }        // 3 bounding boxes in one message
+                                };
+                                cmd.data[2] = (unsigned int)((pObj->y*0x7FFF)+0.5)<<16  | (unsigned int)((pObj->x*0x7FFF)+0.5);
+                                cmd.data[3] = (unsigned int)((pObj->h*0x7FFF)+0.5)<<16  | (unsigned int)((pObj->w*0x7FFF)+0.5);
+                                cmd.data[4] = (unsigned int) ((pObj->confidence*0x3FFFFFFF)+0.5);    // convert float to Q30.1
+                                cmd.data[5] = pObj->label;
+                                pObj++;
+                                if (numBB_per_message > 1) {
+                                    cmd.data[6] = (unsigned int)((pObj->y*0x7FFF)+0.5)<<16  | (unsigned int)((pObj->x*0x7FFF)+0.5);
+                                    cmd.data[7] = (unsigned int)((pObj->h*0x7FFF)+0.5)<<16  | (unsigned int)((pObj->w*0x7FFF)+0.5);
+                                    cmd.data[8] = (unsigned int) ((pObj->confidence*0x3FFFFFFF)+0.5);    // convert float to Q30.1
+                                    cmd.data[9] = pObj->label;
+                                    pObj++;
+                                }
+                                if (numBB_per_message > 2) {
+                                    cmd.data[10] = (unsigned int)((pObj->y*0x7FFF)+0.5)<<16  | (unsigned int)((pObj->x*0x7FFF)+0.5);
+                                    cmd.data[11] = (unsigned int)((pObj->h*0x7FFF)+0.5)<<16  | (unsigned int)((pObj->w*0x7FFF)+0.5);
+                                    cmd.data[12] = (unsigned int) ((pObj->confidence*0x3FFFFFFF)+0.5);    // convert float to Q30.1;
+                                    cmd.data[13] = pObj->label;
+                                    pObj++;
+                                }
+                                ERRCHK(sendCommand(sock, cmd, clientName));
+                                ERRCHK(recvCommand(sock, cmd, clientName, INFCOM_CMD_BB_INFERENCE_RESULT));
+                                j += numBB_per_message;
+                            }
+                        }
+                        resultCountAvailable--;
+                    }
+                    bounding_boxes.clear();
+                }
+            }
+        }
+
+        // if not endOfImageRequested, request client to send images
+        if(!endOfImageRequested) {
+            // get number of empty slots in the input queue
+            int imageCountRequested = 0;
+#if INFERENCE_SCHEDULER_MODE == NO_INFERENCE_SCHEDULER
+            imageCountRequested = 1;
+#elif INFERENCE_SCHEDULER_MODE == LIBRE_INFERENCE_SCHEDULER
+            imageCountRequested = MAX_INPUT_QUEUE_DEPTH - inputQ.size();
+#endif
+            if(imageCountRequested > 0) {
```

Comment:
looks like imageCountRequested >0 is true always. If so get rid of the if condition